vulnerabilities/tests/test_openssl.py

```diff
@@ -20,466 +20,35 @@
-                            name="openssl",
-                            version="1.1.1e",
-                        ),
-                    ),
-                    AffectedPackage(
-                        vulnerable_package=PackageURL(
-                            type="generic",
-                            namespace=None,
-                            name="openssl",
-                            version="1.1.1b",
-                        ),
-                        patched_package=PackageURL(
-                            type="generic",
-                            namespace=None,
-                            name="openssl",
-                            version="1.1.1e",
-                        ),
-                    ),
-                    AffectedPackage(
-                        vulnerable_package=PackageURL(
-                            type="generic",
-                            namespace=None,
-                            name="openssl",
-                            version="1.1.1c",
-                        ),
-                        patched_package=PackageURL(
-                            type="generic",
-                            namespace=None,
-                            name="openssl",
-                            version="1.1.1e",
-                        ),
-                    ),
-                    AffectedPackage(
-                        vulnerable_package=PackageURL(
-                            type="generic",
-                            namespace=None,
-                            name="openssl",
-                            version="1.1.1d",
-                        ),
-                        patched_package=PackageURL(
-                            type="generic",
-                            namespace=None,
-                            name="openssl",
-                            version="1.1.1e",
-                        ),
-                    ),
-                ],
-                references=[
-                    Reference(
-                        reference_id="",
-                        url="https://github.com/openssl/openssl/commit/419102400a2811582a7a3d4a4e317d72e5ce0a8f",
-                        severities=[],
-                    ),
-                    Reference(
-                        reference_id="",
-                        url="https://github.com/openssl/openssl/commit/f1c5eea8a817075d31e43f5876993c6710238c98",
-                        severities=[],
-                    ),
-                ],
-            ),
-        ]
-
-        found_advisories = OpenSSLImporter.to_advisories(data)
-
-        found_advisories = list(map(Advisory.normalized, found_advisories))
-        expected_advisories = list(map(Advisory.normalized, expected_advisories))
-        assert sorted(found_advisories) == sorted(expected_advisories)
+TEST_DATA = os.path.join(BASE_DIR, "test_data", "openssl")
+
+
+class TestOpenssl(unittest.TestCase):
+    def test_to_advisory(self, regen=False):
```

Comment:
yes

Reply:
This sounds weird. How does the regenerate work ?

Reply:
This is borrowed from Tushar's suggestion
https://github.com/nexB/vulnerablecode/pull/648#discussion_r832183736

Reply:
Interesting. Is there a documentation on this ?

Reply:
I don't think we have one for this.

Reply:
This is not good. I don't think having undocumented and unaccounted code is a good idea.

Reply:
Agreed, but this is a common pattern that I use quite often in ScanCode. Adding a simple comment or docstring that documents how to use this would be great and enough IMHO.

Reply:
@Hritik14 will add some more unit tests there, I guess that will justify the file name.